src/common_subset.rs

```diff
@@ -132,7 +151,7 @@ impl<NodeUid: Clone + Debug + Eq + Hash + Ord> CommonSubset<NodeUid> {
         // Create all agreement instances.
         let mut agreement_instances: BTreeMap<NodeUid, Agreement<NodeUid>> = BTreeMap::new();
         for uid0 in all_uids {
-            agreement_instances.insert(uid0.clone(), Agreement::new(uid0.clone(), num_nodes));
+            agreement_instances.insert(uid0.clone(), Agreement::new(uid.clone(), num_nodes));
```

Comment:
I'm interested in how the change works. `uid0` is the proposer ID ranging over all nodes. `uid` is the sender ID and is the ID of the node the ACS instance is set up to run on. If you create a new Agreement each time with the proposer ID `uid`, all the Agreement instances will be associated with this node and none with any remote nodes. Can you please give some insight of why this might not be the case?

Reply:
The first argument of `Agreement::new` is the node's own ID. The `Agreement` algorithm itself has no notion of a proposer, since every node can propose a boolean value.

The proposer ID exists only on the `CommonSubset` level and defines what the corresponding agreement instance is _about_, i.e. whose item's inclusion into the subset we are trying to agree about.

Reply:
Right! Thanks for reminding about that. I still mix up the two roles of IDs sometimes. Possibly we should say either `proposer_id` or `sender_id` instead of just `uid`. The latter is so ambiguous.

Reply:
Good idea, I changed it.